src/HttpServer/HttpServer.cc

```diff
@@ -461,6 +580,8 @@ bool HttpServer::WriteObjectFile(const std::string& object_type, const std::stri
             obj["$schema"] = CARTA_LAYOUT_SCHEMA_URL;
         } else if (object_type == "snippet") {
             obj["$schema"] = CARTA_SNIPPET_SCHEMA_URL;
+        } else if (object_type == "workspace") {
+            obj["$schema"] = CARTA_WORKSPACE_SCHEMA_URL;
```

Comment:
Perhaps this should be a lookup in a `SCHEMA_URL` unordered map. I think you won't need an explicit hash function if you change the strings to enum values (since C++14 `std::hash` can be used on enums directly, so I believe that the default Just Works now).